src/analytics/private/sqls/redshift/sp-scan-metadata.sql

```diff
@@ -278,70 +280,54 @@ BEGIN
 				property_value, 
 				value_type, 
 				parameter_count,
+				event_name,
 				platform
-			FROM (
-				SELECT
-					event_name,
+			FROM (						
+				SELECT 
 					project_id, 
 					app_info_app_id, 
-					property_category,
-					month, 
-					day_number, 
-					property_name, 
-					property_value, 
-					value_type, 
-					parameter_count,
+					property_category, 
+					month,
+					day_number,
+					value_type,
+					property_name,
+					property_value,
+					event_name,
 					platform,
-					ROW_NUMBER() OVER (PARTITION BY event_name, project_id, app_info_app_id, property_category, month, day_number, property_name, value_type, platform ORDER BY parameter_count DESC) AS row_num
+					parameter_count,
+					ROW_NUMBER() OVER (PARTITION BY project_id, app_info_app_id, property_category, month, day_number, value_type, property_name ORDER BY parameter_count DESC) AS row_num
 				FROM (
-					SELECT 
+					SELECT
 						event_name, 
 						project_id, 
 						app_info_app_id, 
 						property_category, 
-						month,
-						day_number,
+						'#' || TO_CHAR(event_date::DATE, 'YYYYMM') AS month,
+						TO_CHAR(event_date::DATE, 'DD')::INTEGER AS day_number,
 						property_name,
 						property_value, 
-						value_type,
-						LISTAGG(platform, '#|!|#') WITHIN GROUP (ORDER BY platform) AS platform, 
-						SUM(parameter_count) AS parameter_count 
-					FROM (
-						SELECT
-							event_name, 
-							project_id, 
-							app_info_app_id, 
-							property_category, 
-							'#' || TO_CHAR(event_date::DATE, 'YYYYMM') AS month,
-							TO_CHAR(event_date::DATE, 'DD')::INTEGER AS day_number,
-							property_name,
-							property_value, 
-							value_type, 
-							platform,
-							count(*) AS parameter_count
-						FROM
-							properties_temp_table
-						WHERE 
-							property_value IS NOT NULL AND
-							property_value != ''
-						GROUP BY event_name, project_id, app_info_app_id, property_category, month, day_number, property_name, property_value, value_type, platform									
-					)
-					GROUP BY event_name, project_id, app_info_app_id, property_category, month, day_number, property_name, property_value, value_type
+						value_type, 
+						platform,
+						count(*) AS parameter_count
```

Comment:
Try to use `HyperLogLog` to optimize the performance in large dataset

https://docs.aws.amazon.com/redshift/latest/dg/hyperloglog-overview.html

Reply:
Currently, just use group by to count parameter count. The performance is fine.